### PR TITLE
Make issuer CA cert check optional

### PR DESCRIPTION
**Pull request by vladimir-mencl-eresearch**
Hi Bradley,

As discussed, this is the feature to make the issuer CA cert check optional (but defaulting to the current behaviour even with untouched config file, as well as with new default config file).

Plus a few fixes I picked up on the way.

Please let me know if OK to merge.

Cheers,
Vlad

### CODE REVIEW

File: app/plugins/base/grails-app/services/aaf/fr/foundation/CryptoService.groovy [hunk 1]
```diff
@@ -9,10 +10,19 @@ import java.security.cert.*
  *
  * @author Bradley Beddoes
  */
-class CryptoService {
+class CryptoService implements InitializingBean {
```
Comment (bradleybeddoes) [inline app/plugins/base/grails-app/services/aaf/fr/foundation/CryptoService.groovy]: When building this what was the motivation for using this bean over the alternative of having Grails do injection? I have no issue with this approach per se but if we can get away with something more minimal then I'd be +1 to that.

e.g. I was thinking something like the following might be possible:

``` groovy
class CryptoService
  def grailsApplication

  ..

  def boolean validateCertificate(aaf.fr.foundation.Certificate certificate, boolean requireChain) {
   ignoreIssuerCA = grailsApplication.config.aaf.fr.certificates.ignoreIssuerCA
   log.debug "Validating certificate ${certificate.subject} with issuer ${certificate.issuer}"	
   if(!requireChain && ( (certificate.subject == certificate.issuer) || ignoreIssuerCA ) ) {
     log.debug "requireChain is false and cert is either self signed or we ignore CA, valid."
     return true
  }

 ..

}
```

Comment (vladimir-mencl-eresearch) [inline app/plugins/base/grails-app/services/aaf/fr/foundation/CryptoService.groovy]: Hi Bradley,

I thought you would not have liked the use of grailsApplication.config in the actual service method and therefore I pre-populated a local variable.

I'd also think of your way as being cleaner - I'll push an update shortly.

Cheers,
Vlad

**Comment by vladimir-mencl-eresearch** [comment]
Hi Bradley, how's it looking now...?

**Comment by bradleybeddoes** [comment]
Much cleaner, thanks!

**Pull request merged**